src/tribol/mesh/MfemData.cpp

```diff
@@ -670,34 +684,41 @@ const MfemSubmeshData::UpdateData& MfemSubmeshData::GetUpdateData() const
 
 MfemJacobianData::MfemJacobianData( const MfemMeshData& parent_data, const MfemSubmeshData& submesh_data,
                                     ContactMethod contact_method )
-    : parent_data_{ parent_data }, submesh_data_{ submesh_data }, block_offsets_( 3 )
+    : parent_data_{ parent_data }, submesh_data_{ submesh_data }, block_offsets_( 3 ), disp_offsets_( 2 )
 {
   SLIC_ERROR_ROOT_IF( parent_data.GetParentCoords().ParFESpace()->FEColl()->GetOrder() > 1,
                       "Higher order meshes not yet supported for Jacobian matrices." );
 
-  // NOTE: Looks like GetFrom() should be const in MFEM
-  mfem::SubMeshUtils::BuildVdofToVdofMap( parent_data_.GetSubmeshFESpace(), *parent_data_.GetParentCoords().FESpace(),
-                                          const_cast<mfem::ParSubMesh&>( parent_data_.GetSubmesh() ).GetFrom(),
-                                          parent_data_.GetSubmesh().GetParentElementIDMap(),
-                                          submesh2parent_vdof_list_ );
+  mfem::SubMeshUtils::BuildVdofToVdofMap(
+      parent_data_.GetSubmeshFESpace(), *parent_data_.GetParentCoords().FESpace(), parent_data_.GetSubmesh().GetFrom(),
+      parent_data_.GetSubmesh().GetParentElementIDMap(), submesh2parent_vdof_list_ );
 
-  int my_rank{ 0 };
-  MPI_Comm_rank( TRIBOL_COMM_WORLD, &my_rank );
-  auto offset_idx = HYPRE_AssumedPartitionCheck() ? 0 : my_rank;
-  auto dof_offset = parent_data_.GetParentCoords().ParFESpace()->GetDofOffsets()[offset_idx];
+  auto dof_offset = parent_data_.GetParentCoords().ParFESpace()->GetMyDofOffset();
   for ( auto& vdof : submesh2parent_vdof_list_ ) {
     vdof = vdof + dof_offset;
   }
 
+  auto& parent_fes = *parent_data_.GetParentCoords().ParFESpace();
```

Comment:
Is this new? I can't tell in github if this code was always here and just moved. Can you clarify and add a comment as to what this block is doing?

Reply:
Yeah, it creates an operator to do the parent<->submesh map.  I want to transition the old mortar method to do this, since it cleans up code, but we can do this in a follow on.